pkg/controller/gittrackobject/gittrackobject_controller_test.go

```diff
@@ -511,348 +537,474 @@ var _ = Describe("GitTrackObject Suite", func() {
+			Context("and ClusterGitTrackMode set to IncludeNamespaced and GitTrackMode set to disabled", func() {
+				var gto *farosv1alpha1.GitTrackObject
+				var namespacedChild *appsv1.Deployment
+				var originalVersion string
+
+				BeforeEach(func() {
+					SetupTest(farosflags.GTMDisabled, farosflags.CGTMIncludeNamespaced)
 				})
+				BeforeEach(func() {
+					gto = testutils.ExampleGitTrackObject.DeepCopy()
+					gto.SetOwnerReferences([]metav1.OwnerReference{
+						{
+							APIVersion: "faros.pusher.com/v1alpha1",
+							Kind:       "ClusterGitTrack",
+							UID:        clusterGitTrack.UID,
+							Name:       clusterGitTrack.Name,
+						},
+					})
+					namespacedChild = testutils.ExampleDeployment.DeepCopy()
+					Expect(testutils.SetGitTrackObjectInterfaceSpec(gto, namespacedChild)).To(Succeed())
+					m.Create(namespacedChild).Should(Succeed())
+					// get the most recent version so we can get the UID and version
+					m.Get(namespacedChild, timeout).Should(Succeed())
+					originalUID = namespacedChild.GetUID()
+					originalVersion = namespacedChild.GetResourceVersion()
 
-				Context("when the child has the update strategy", func() {
-					var originalVersion string
-					var originalUID types.UID
+					// Create
+					m.Create(gto).Should(Succeed())
+				})
+
+				It("should reconcile namespaced GitTrackObjects owned by ClusterGitTracks", func() {
+					Eventually(requests, timeout).Should(Receive(Equal(expectedRequest)))
+				})
+				It("should update the child", func() {
+					m.Eventually(namespacedChild, timeout).ShouldNot(testutils.WithResourceVersion(Equal(originalVersion)))
+				})
+			})
+
+			Context("and ClusterGitTrackMode set to IncludeNamespaced", func() {
+				BeforeEach(func() {
+					SetupTest(farosflags.GTMEnabled, farosflags.CGTMIncludeNamespaced)
+				})
 
+				BeforeEach(func() {
+					gto = testutils.ExampleClusterGitTrackObject.DeepCopy()
+					gto.SetOwnerReferences([]metav1.OwnerReference{
+						{
+							APIVersion: "faros.pusher.com/v1alpha1",
+							Kind:       "ClusterGitTrack",
+							UID:        clusterGitTrack.UID,
+							Name:       clusterGitTrack.Name,
+						},
+					})
+					child = testutils.ExampleClusterRoleBinding.DeepCopy()
+					Expect(testutils.SetGitTrackObjectInterfaceSpec(gto, child)).To(Succeed())
+				})
+
+				Context("with valid data", func() {
 					BeforeEach(func() {
-						child.SetAnnotations(map[string]string{"updated": "annotations"})
-						child.SetOwnerReferences([]metav1.OwnerReference{testutils.GetClusterGitTrackObjectOwnerRef(gto)})
-						m.Update(child).Should(Succeed())
-						// Wait for the update reconcile
+						// Create and fetch the instance to make sure caches are synced
+						m.Create(gto).Should(Succeed())
+						// Wait for the initial reconcile
 						Eventually(requests, timeout).Should(Receive(Equal(expectedClusterRequest)))
-						// And for the status update reconcile
+						// And for the status one as well, probably
```

Comment:
I don't like adding `probably` here, since the next line is a `definitely` should receive an update for the status update reconcile request

Reply:
I am confused. I did not add this comment and I suspect that this is from the diff algorithm